apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx

```diff
@@ -57,11 +65,65 @@ const analyticsEditEventProps = {
   button: "Edit",
 };
 
+const TabContainer = styled(Flex).attrs({
+  mx: 11,
+  mt: 6,
+  p: 1,
+  columnGap: 1,
+  borderRadius: "9px",
+  backgroundColor: "neutral.c30",
+  flexDirection: "row",
+  flexGrow: 0,
+  alignSelf: "center",
+})``;
+
+function Tab({
+  isActive,
+  onPress,
+  children,
+}: {
+  isActive: boolean;
+  onPress(): void;
+  children: string;
+}): React.JSX.Element {
+  return (
+    <Pressable onPress={onPress} style={{ width: "50%" }}>
+      <Flex backgroundColor={isActive ? "neutral.c50" : "transparent"} borderRadius={"8px"} p={3}>
+        <Text
+          variant={"paragraph"}
+          fontWeight={"semiBold"}
+          color={isActive ? "palette.neutral.c100" : "palette.neutral.c70"}
+          textAlign={"center"}
+        >
+          {children}
+        </Text>
+      </Flex>
+    </Pressable>
+  );
+}
+
 const PreviewPreEdit = ({ navigation, route }: NavigationProps) => {
   const { t } = useTranslation();
   const [loadedImage, setLoadedImage] = useState<ImageFileUri | null>(null);
   const { params } = route;
   const { isPictureFromGallery, device, isStaxEnabled } = params;
+  const [deviceModelId, setSelectedDeviceModelId] = useState<CLSSupportedDeviceModelId>(
+    params.deviceModelId ?? DeviceModelId.stax,
+  );
+
+  const supportDeviceStax = useFeature("supportDeviceStax")?.enabled;
+  const supportDeviceEuropa = useFeature("supportDeviceEuropa")?.enabled;
+  const supportedAndEnabledDeviceModelIds = supportedDeviceModelIds.filter(() => {
+    const devicesSupported: Record<CLSSupportedDeviceModelId, boolean> = {
+      [DeviceModelId.stax]: Boolean(supportDeviceStax),
+      [DeviceModelId.europa]: Boolean(supportDeviceEuropa),
+    };
```

Comment:
There is one entry point where the device model is undetermined: the `ledgerlive://custom-image` deeplink. It's used in a QR code that's baked in the firmware so I think the easiest is just to handle the logic here.
If a Stax user arrives there while Europa is not released yet (and the associated flag is off), it will be problematic to display Europa.
We could also check with the firmware if it's possible to add a query parameter `deviceModelId` in the deeplink.

Reply:
✅ 

didn't think about deeplink.